r/R/worker_helpers.R

```diff
@@ -143,11 +143,44 @@ getClusters <- function(type, resolution, data) {
   return(data)
 }
 
-handle_pagination <- function(gene_results, offset, limit, order_by, order_decreasing, filter) {
-  if (!is.null(filter)) {
-    gene_results <- gene_results[grepl(filter, gene_results$gene_name, ignore.case = TRUE), ]
+applyFilters <- function(gene_results, filters) {
+
+  filter_columns <- sapply(filters, `[[`, 'columnName')
+
+  # apply gene text filter
+  gene.idx <- which(filter_columns == 'gene_names')[1]
+  if (!is.na(gene.idx)) {
+    gene <- filters[[gene.idx]]$expression
```

Comment:
It took me a second to figure out that expression referred to regex. Maybe we change it? It's pretty clear what it does, but just for clarity.

Whatever you think is fine.

Reply:
`expression` is what the UI is sending as a request. Agree I'm not a fan